### PR TITLE
Ensure NominationProtocol initialization & Add std::set.empty() support

### PR DESCRIPTION
**Pull request by AndrejMitrovic**

### CODE REVIEW

File: source/scpd/Cpp.d [hunk 1]
```diff
@@ -104,6 +107,12 @@ extern(C++, `std`) {
             return cpp_set_foreach!Key(cast(void*)&this, cast(void*)&dg,
                 cast(void*)&wrapper);
         }
+
+        /// Returns: true if the set is empty
+        extern(D) bool empty ()
```
Comment (Geod24) [inline source/scpd/Cpp.d]: `const` ?

**Comment by AndrejMitrovic** [comment]
Added keyword soup.

File: source/scpd/Cpp.d [hunk 2]
```diff
@@ -104,6 +107,12 @@ extern(C++, `std`) {
             return cpp_set_foreach!Key(cast(void*)&this, cast(void*)&dg,
                 cast(void*)&wrapper);
         }
+
+        /// Returns: true if the set is empty
+        extern(D) bool empty () const nothrow pure @nogc
+        {
+            return cpp_set_empty!Key(cast(void*)&this);
```
Comment (Geod24) [inline source/scpd/Cpp.d]:
```suggestion
            return cpp_set_empty!Key(cast(const void*)&this); 1 
```

File: source/scpp/extra/DUtils.h [hunk 3]
```diff
@@ -24,6 +24,11 @@ int cpp_set_foreach(void* setptr, void* ctx, void* func)
     return 0;
 }
 
+template<typename T>
+bool cpp_set_empty(void* setptr)
+{
+    return ((std::set<T>*)setptr)->empty();
```
Comment (Geod24) [inline source/scpp/extra/DUtils.h]: Ditto `const`

File: source/scpp/extra/DUtils.h [hunk 4]
```diff
@@ -24,6 +24,11 @@ int cpp_set_foreach(void* setptr, void* ctx, void* func)
     return 0;
 }
 
+template<typename T>
+bool cpp_set_empty(void* setptr)
```
Comment (Geod24) [inline source/scpp/extra/DUtils.h]: And here

**Comment by AndrejMitrovic** [comment]
Made it more const correct.

**Review by Geod24 — APPROVED** [review]

**Pull request merged**